client/server/pages/index.js

```diff
@@ -785,30 +788,31 @@ module.exports = function() {
 	} );
 
 	// Landing pages for domains-related emails
-	app.get( '/domain-services/:action', function( req, res ) {
-		const ctx = getDefaultContext( req, 'entry-domains-landing' );
-		attachBuildTimestamp( ctx );
-		attachHead( ctx );
-		attachI18n( ctx );
-
-		ctx.clientData = config.clientData;
-		ctx.domainsLandingData = {
-			action: get( req, [ 'params', 'action' ], 'unknown-action' ),
-			query: get( req, 'query', {} ),
-		};
-
-		const pageHtml = renderJsx( 'domains-landing', ctx );
-		res.send( pageHtml );
-	} );
+	app.get(
+		'/domain-services/:action',
+		setupDefaultContext( 'entry-domains-landing' ),
+		( req, res ) => {
+			const ctx = req.context;
+			attachBuildTimestamp( ctx );
+			attachHead( ctx );
+			attachI18n( ctx );
+
+			ctx.clientData = config.clientData;
+			ctx.domainsLandingData = {
+				action: get( req, [ 'params', 'action' ], 'unknown-action' ),
+				query: get( req, 'query', {} ),
+			};
+
+			const pageHtml = renderJsx( 'domains-landing', ctx );
+			res.send( pageHtml );
+		}
+	);
 
 	function handleSectionPath( section, sectionPath, entrypoint ) {
 		const pathRegex = pathToRegExp( sectionPath );
 
-		app.get( pathRegex, function( req, res, next ) {
-			req.context = {
-				...getDefaultContext( req, entrypoint ),
-				sectionName: section.name,
-			};
+		app.get( pathRegex, setupDefaultContext( entrypoint ), function( req, res, next ) {
+			req.context.sectionName = section.name;
```

Comment:
Any concerns with the fact that we're now mutating `req.context` while before we were creating a new object?

I can see it's the same case for `/browsehappy` too.

Reply:
In this case, the `req.context` object is designed to be mutated as it passes through the handler chain. Treating it as immutable is not necessary and only adds inefficiencies when cloning the object.

The only danger is the case where the `req.context` wasn't initialized by any previous handler. Then:
```js
req.context = { ...req.context, foo: 'bar' };
```
initializes it without blinking (spreading a `null` value is OK), but:
```js
req.context.foo = 'bar';
```
would crash.

Reply:
Makes sense 👍